File: examples/typescript/04-handle-rate-limit.ts
// Catch a RateLimitError and inspect the rate-limit headers.
import { BimpeAI, RateLimitError } from '@bimpeai/sdk';

const client = new BimpeAI({ apiKey: process.env.BIMPEAI_API_KEY ?? '', maxRetries: 0 });

try {
  await client.agents.list();
} catch (error) {
  if (error instanceof RateLimitError) {
    console.log(
      `rate limited. retry in ${error.retryAfter}s, limit ${error.limit}, remaining ${error.remaining}`,
    );
  } else {
    throw error;
  }
}

File: packages/sdk-python/examples/04_handle_rate_limit.py
import os

from bimpeai import BimpeAI, RateLimitError

client = BimpeAI(api_key=os.environ.get("BIMPEAI_API_KEY", ""), max_retries=0)
try:
    client.agents.list()
except RateLimitError as err:
    print(f"rate limited; retry in {err.retry_after}s, remaining {err.remaining}")
